diffrax/integrate.py

```diff
@@ -204,13 +315,51 @@ def _apply_history(x):
         tprev = jnp.minimum(tprev, t1)
         tnext = _clip_to_end(tprev, tnext, t1, keep_step)
 
+        # updating discontinuity
+        jumped = lambda a, b: jnp.where(discont_made_jump, a, b)
+        discontinuity = state.discontinuity
+        discontinuity_save_index = state.discontinuity_save_index
+        if delays is not None:
+            discontinuity = jumped(
+                discontinuity.at[discontinuity_save_index].set(tnext),
+                discontinuity,
+            )
+            discontinuity_save_index = jumped(
+                discontinuity_save_index + 1, discontinuity_save_index
+            )
+
+            # Check if discontinuity is very close to state.tprev in order
+            # to give it a little push to skip discontinuity
+            pred = jnp.isclose(
+                discontinuity[discontinuity_save_index - 1],
+                tprev,
+            )
+            tprev = jnp.where(pred, nextafter(nextafter(tprev)), tprev)
+            # If the last step we move back a little bit in order
+            # to stop right before discontinuity
+            tnext = jnp.where(discont_made_jump, prevbefore(tnext), tnext)
+
+            # Edge case where we integration from state.tprev to state.tnext
+            # and future state.tprev is infinitely close
+            # to the next discontinuity hence we are integration on a
+            # very small interval and how things
+            # are right now this situation isn't dealt with properly so this
+            # are work around, can also put
+            # tnext + (state.ts[state.save_index ]- state.ts[state.save_index-1])
+            # instead of tnext + 0.01
+            tnext = jnp.where(
+                discont_made_jump & pred,
+                tnext + (state.ts[state.save_index] - state.ts[state.save_index - 1]),
+                tnext,
+            )
+
```

Comment:
This is the edge case, I was talking about.

Reply:
(For completeness -- I've seen this but I'm a little short on time right now. I'll leave you with all the other comments I've written and cover this in another round of review later. But definitely better to fix edge cases before I review them anyway ;) )